tests/extendrtests/src/rust/src/ndarray.rs

```diff
@@ -0,0 +1,33 @@
+use extendr_api::prelude::*;
+
+/// Calculate Euclidean distance matrix
+/// Test case adopted from https://github.com/mikemahoney218/examplerust/blob/23d21b1ced4e24b7a7c00dd36290114dc1bbd113/src/rust/src/lib.rs#L5
+/// @param a : Matrix of real values or `NULL`
+/// @export
+#[extendr(use_try_from = true)]
+fn euclidean_dist(a: Nullable<ArrayView2<Rfloat>>) -> Nullable<Doubles> {
+    if let NotNull(a) = a {
+        let nrow = a.nrows();
+
+        let result = (0..(nrow - 1))
```

Comment:
May just iter `a.rows()` for a clearer intent?

Reply:
I copy-pasted an example provided by one of the users (see github link) and updated it so that it does not use types other than extendr types (like `Rfloat`). In this case it uses current row number to execute inner loop, and I see no simple solution here.